prometheus/testutil/testutil.go

```diff
@@ -108,6 +108,34 @@ func ToFloat64(c prometheus.Collector) float64 {
 	panic(fmt.Errorf("collected a non-gauge/counter/untyped metric: %s", pb))
 }
 
+// MetricCount collects all Metrics from the provided Collector and counts them.
+//
+// Typical use case is when you want to assert certain number of series of given collector after
+// some operation.
+//
+// This function is only for testing purposes, and even for testing, other approaches
+// are often more appropriate (see this package's documentation).
+func MetricCount(c prometheus.Collector) int {
```

Comment:
Perhaps call it `CollectAndCount` so that it corresponds to `CollectAndCompare`?